src/file_integrity.rs

```diff
@@ -106,12 +109,19 @@ impl FileIntegrity {
     }
 
     /// Checking if EVERY file is actully matching the expected checksum
-    pub fn check(&self) -> bool {
+    pub fn check(&self, report: &mut Report) -> bool {
         let mut res = true;
```

Comment:
This should probably return a `Result` if possible. It could return an 'error' that contains a list of the violating files. e.g. `Result<(), Vec<PathBuf>>`

Boolean values in Rust are generally frowned upon as they aren't perfectly specific about what they return. Again, this isn't extremely scary for this project, but in production Rust code, many users [swear by the type system](https://blakesmith.me/2019/05/07/rust-patterns-enums-instead-of-booleans.html).